sys/fuchsia/fidl_process.txt

```diff
@@ -1,13 +1,90 @@
-# Copyright 2018 syzkaller project authors. All rights reserved.
-# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+# Copyright 2018 The Fuchsia Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+#
+# WARNING: This file is machine generated by fidlgen.
 
-include <zircon/syscalls.h>
+fuchsia_process_LauncherLaunchRequest {
+	txid		int32[0]
+	reserved	int32[0]
+	flags		int32[0]
+	ordinal		int32[1]
+	infoInLine	fuchsia_process_LaunchInfoInLine
+	infoOutOfLine	fuchsia_process_LaunchInfoOutOfLine
+} [align_8, packed]
 
-resource zx_chan_fuchsia_process_Launcher_client[zx_chan]
-resource zx_chan_fuchsia_process_Launcher_server[zx_chan]
+fuchsia_process_LaunchInfoInLine {
+	executable	flags[fidl_handle_presence, int32]
+	job		flags[fidl_handle_presence, int32]
+	nameInLine	fidl_string
+} [align_8, packed]
```

Comment:
Both `align_8` and `packed` look wrong here and on most other structs.
According to the doc:
https://fuchsia.googlesource.com/zircon/+/master/docs/fidl/wire-format/index.md#structs
struct alignment is maximum alignment of its fields, which is exactly the C and syzkaller default rules. If a struct contains just `int8` its alignment should be 1, with `align_8` syzkaller will lay it out incorrectly.
Similarly, `packed` will prevent addition of any alignment packing, but fidl rules require alignment padding. And again fidl rules match C and syzkaller default rules.
So it seems to me we just need to remove both attributes.
However, the out-of-line data needs to be aligned to 8 as a whole (?).

Reply:
The wire-format seems to require per-element-basis natural alignment. So I removed all the alignment requirement for composite data structures. If needed, each field should have their own alignment restriction, just like you suggested one for string.